incubator/patroni/charts/etcd/templates/etcd-petset.yaml

```diff
@@ -0,0 +1,212 @@
+  ports:
+  - port: {{.Values.PeerPort}}
+    name: etcd-server
+  - port: {{.Values.ClientPort}}
+    name: etcd-client
+  clusterIP: None
+  selector:
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+---
+apiVersion: apps/v1alpha1
+kind: PetSet
+metadata:
+  name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  labels:
+    heritage: {{.Release.Service | quote }}
+    release: {{.Release.Name | quote }}
+    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    component: "{{.Release.Name}}-{{.Values.Component}}"
+  annotations:
+    "helm.sh/created": {{.Release.Time.Seconds | quote }}
+spec:
+  serviceName: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+  replicas: {{default 3 .Values.Replicas | quote }}
+  template:
+    metadata:
+      name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+      labels:
+        heritage: {{.Release.Service | quote }}
+        release: {{.Release.Name | quote }}
+        chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+        component: "{{.Release.Name}}-{{.Values.Component}}"
+      annotations:
+        "helm.sh/created": {{.Release.Time.Seconds | quote }}
+        pod.alpha.kubernetes.io/initialized: "true"
+    spec:
+      containers:
+      - name: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+        image: "{{.Values.Image}}:{{.Values.ImageTag}}"
+        imagePullPolicy: "{{.Values.ImagePullPolicy}}"
+        ports:
+        - containerPort: {{.Values.PeerPort}}
+          name: peer
+        - containerPort: {{.Values.ClientPort}}
+          name: client
+        resources:
+          requests:
+            cpu: "{{.Values.Cpu}}"
+            memory: "{{.Values.Memory}}"
+        env:
+        - name: INITIAL_CLUSTER_SIZE
+          value: {{default 3 .Values.Replicas | quote }}
+        - name: PETSET_NAME
+          value: "{{ printf "%s-%s" .Release.Name .Values.Name | trunc 24 }}"
+        volumeMounts:
+        - name: datadir
+          mountPath: /var/run/etcd
+        lifecycle:
+          preStop:
+            exec:
+              command:
+                - "/bin/sh"
+                - "-ec"
+                - |
+                  EPS=""
+                  for i in $(seq 0 $((${INITIAL_CLUSTER_SIZE} - 1))); do
+                      EPS="${EPS}${EPS:+,}http://${PETSET_NAME}-${i}.${PETSET_NAME}:2379"
+                  done
+
+                  HOSTNAME=$(hostname)
+
+                  member_hash() {
+                      etcdctl member list | grep http://${HOSTNAME}.${PETSET_NAME}:2380 | cut -d':' -f1 | cut -d'[' -f1
+                  }
+
+                  echo "Removing ${HOSTNAME} from etcd cluster"
+
+                  ETCDCTL_ENDPOINT=${EPS} etcdctl member remove $(member_hash)
+                  if [ $? -eq 0 ]; then
+                      # Remove everything otherwise the cluster will no longer scale-up
+                      rm -rf /var/run/etcd/*
+                  fi
+        command:
+          - "/bin/sh"
```

Comment:
Yes for now vendoring is the right approach but the start of chart dependency management was merged a few days ago and will be in helm alpha.5 :
https://github.com/kubernetes/helm/pull/1150